### PR TITLE
Frontend/store

### LINKED ISSUE #497: [Frontend] Add store management to the app to decouple code
As a developer, I want a global store management to allow me to access loaded data on any components without directly passing them down from the parents

### PR DESCRIPTION
**Pull request by ds161833**
Implemented a store object as a context, now if a component needs dimensions data they can get it directly from the store instead of accepting it from a waterfall of props

### CODE REVIEW

File: Client/src/Components/DatasetUpload/DataSection/VariableHeader.tsx [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
   variable: IVariable,
   index: number,
   onHeaderClick: (index: number) => void
-  dimensions: IDimensionModel[]
 }
 
 export const VariableHeader = (props: IProps) => {
-  const { variable, index, onHeaderClick, dimensions } = { ...props }
+  const { variable, index, onHeaderClick } = { ...props }
+  const { dimensions } = useContext(StoreContext).store.getPreloadedData()
```
Comment (GD-Dheer) [inline Client/src/Components/DatasetUpload/DataSection/VariableHeader.tsx:16]: Much cleaner

File: Client/src/Context/UserContext.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  setUserContext: (user: IUserAccountModel) => void
+}
+
+interface IProps {
+  children: any
+}
+
+export const UserContext = React.createContext<Partial<IUserContextProps>>({})
+
+export const UserContextWrapper = (props: IProps) => {
```
Comment (GD-Dheer) [inline Client/src/Context/UserContext.tsx:17]: Could you add this to the root? Similar call as variables & dimensions and lower overhead. We wrap app.ts with 2 different context at the moment

File: Client/src/Context/StoreContext.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  store: RootStore
+}
+
+interface IProps {
+  children: any
+}
+
+export const StoreContext = React.createContext<Partial<IStoreContextProps>>({})
+
+export const StoreContextWrapper = (props: IProps) => {
```
Comment (GD-Dheer) [inline Client/src/Context/StoreContext.tsx:15]: I like this wrapper style component, nicely done

File: Client/src/Components/Utils/Stores/RootStore.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+export class RootStore {
+  dimensionsStore: DimensionsStore
+  variablesStore: VariablesStore
+
+  constructor() {
+    this.dimensionsStore = new DimensionsStore()
+    this.variablesStore = new VariablesStore()
+  }
+
+  getPreloadedData(): IStore {
```
Comment (GD-Dheer) [inline Client/src/Components/Utils/Stores/RootStore.tsx:20]: We can use this to handle showing # of datasets too!

**Review by GD-Dheer — APPROVED** [review]
This is a pretty significant improvement to our current way of accessing information. Saves us prop drilling at least in some places. Great work

File: Client/src/Components/Graph/GraphView.tsx [hunk 5]
```diff
@@ -20,19 +19,11 @@ export default function GraphView() {
...
 
-  const getDimensions = async () => {
-    const databaseDimensions = await callGetAllDimensions()
-    setDimensions(databaseDimensions)
-  }
-
-  useEffect(() => {
-    getDimensions()
-  }, [])
+  useEffect(() => { store.dimensionsStore.loadDimensions() }, [])
```
Comment (laithAwad) [inline Client/src/Components/Graph/GraphView.tsx:26]: much cleaner

File: Client/src/Components/DatasetUpload/DataSection/VariableHeader.tsx (hunk 1 above)
Comment (ds161833) [reply Client/src/Components/DatasetUpload/DataSection/VariableHeader.tsx:16]: yeah we can just store the user in the store, didnt do it cause felt like too many changes already, but I'll do it

**Review by laithAwad — APPROVED** [review]
Great work! this makes it much easier to use preloaded data and organizes our files

File: Client/src/Components/DatasetUpload/DataSection/VariableHeader.tsx (hunk 1 above)
Comment (ds161833) [reply Client/src/Components/DatasetUpload/DataSection/VariableHeader.tsx:16]: will implement this in the pr right after cause it is a lot of new changes that will get mixed up with the changes of this PR

**Pull request merged**